core/src/style/textStyleBuilder.cpp

```diff
@@ -462,14 +554,12 @@ TextStyle::Parameters TextStyleBuilder::applyRule(const DrawRule& _rule,
 
     TextStyle::Parameters p;
 
-    _rule.get(StyleParamKey::text_source, p.text);
-    if (!_rule.isJSFunction(StyleParamKey::text_source)) {
```

Comment:
If the text is from a function the value of `text_source` is a string otherwise it's `StyleParam::TextSource`.

https://github.com/tangrams/tangram-es/pull/1349/files/110e1d9264bea34e2e73cae59bd3b7d5d62c45e7#diff-3f0d16e06da67d94777462bfd1445c27R171